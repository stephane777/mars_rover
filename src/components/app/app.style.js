import styled from "styled-components";

export const AppWrapper = styled.main`
	min-height: 80vh;
	max-width: 120rem;
	margin: 0 auto;
	display: flex;
	flex-direction: column;
	align-items: center;
`;

export const ErrorMessage = styled.p`
	color: var(--alert);
	background-color: var(--bg-alert);
	width: 40rem;
	font-size: 1.4rem;
	text-align: center;
	padding: 1rem;
	border-radius: 6px;
	border-color: var(--border-alert);
`;

export const Heading = styled.h1`
	font-size: 3rem;
	font-weight: 700;
	background-image: linear-gradient(
		to right,
		var(--color-primary),
		var(--color-mars)
	);

	color: rgba(0, 0, 0, 0.2);
	background-clip: text;
	-webkit-background-clip: text;

	@media only screen and (max-width: 576px) {
		font-size: 2rem;
	}
`;
