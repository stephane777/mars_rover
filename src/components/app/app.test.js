import React from "react";
import "@testing-library/jest-dom";
import {
	render,
	screen,
	act,
	within,
	waitForElementToBeRemoved,
	waitFor,
	cleanup,
	fireEvent,
} from "@testing-library/react";
import { App } from "./app.component";

afterEach(cleanup);

describe("it should render the component", () => {
	it("should render the footer component", () => {
		const { container } = render(<App />);
		expect(container.firstChild.className).toMatch("App");
	});
	it("should show Rover Control Board", () => {
		render(<App />);
		expect(screen.getByRole("heading")).toBeInTheDocument();
		expect(screen.getByText(/rover control board/i)).toBeInTheDocument();
	});
	it("should show Send order to rover", () => {
		render(<App />);
		expect(screen.getByLabelText("Send order to rover:")).toBeInTheDocument();
		expect(screen.getByText(/send order to rover:/i)).toBeInTheDocument();
	});
	it("should show Rover Direction", () => {
		render(<App />);
		expect(screen.getByText(/rover direction/i)).toBeInTheDocument();
	});
	it("should show Rover position", () => {
		render(<App />);
		expect(screen.getByText(/rover position/i)).toBeInTheDocument();
	});
	it("should show a Go! button and disabled", () => {
		render(<App />);
		expect(screen.getByRole("button", { name: /Go!/i })).toBeDisabled();
		expect(screen.getByRole("button", { name: /Go!/i })).toBeInTheDocument();
	});
	it("should show a reset position button", () => {
		render(<App />);
		expect(screen.getByRole("button", { name: /reset/i })).not.toBeDisabled();
		expect(screen.getByRole("button", { name: /reset/i })).toBeInTheDocument();
	});

	it("should contain a grid with 100 cell", () => {
		const arrayOf100 = [...Array(100).keys()];

		render(<App />);
		expect.assertions(100);
		arrayOf100.forEach((cell, i) => {
			const dataTestId = `${i}`.length === 1 ? `0${i}` : i;
			expect(screen.getByTestId(dataTestId)).toBeInTheDocument();
		});
	});

	it("should show the Rover image at position [0,0]", () => {
		render(<App />);
		const cellInitialPosition = document.getElementById("00");
		expect(screen.getByRole("img", { name: /rover/i })).toBeInTheDocument();
		expect(screen.getByRole("img", { name: /rover/i })).toHaveAttribute(
			"alt",
			"rover"
		);
		const img = within(cellInitialPosition).getByRole("img", {
			name: /rover/i,
		});
		expect(img).toBeInTheDocument();
	});

	// it("should move the rover to the right", async () => {
	// 	render(<App />);
	// 	// const cellInitialPosition = document.getElementById("10");

	// 	const goButton = screen.getByRole("button", { name: /Go!/i });
	// 	const input = screen.getByRole("textbox", {
	// 		name: /send order to rover:/i,
	// 	});
	// 	expect(screen.getByTestId(/rover_09/)).toBeInTheDocument();
	// 	act(() => {
	// 		fireEvent.change(input, {
	// 			target: { value: "r" },
	// 		});
	// 		fireEvent.click(goButton);
	// 	});
	// 	await waitFor(() => {
	// 		expect(screen.getByTestId("rover_19")).toBeInTheDocument();
	// 	});
	// 	// await waitForElementToBeRemoved(() => screen.getByTestId(/rover_09/i));
	// 	screen.debug();

	// 	// const rover = await screen.findByTestId(`rover_09`);
	// 	// expect(rover).toBeInTheDocument();
	// });

	describe("it should match the snapshot", () => {
		it("should match snapshot", () => {
			const testUnit = render(<App />);
			expect(testUnit).toMatchSnapshot();
		});
	});
});
