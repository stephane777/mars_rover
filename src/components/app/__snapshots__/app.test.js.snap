// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the component it should match the snapshot should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="sc-bdvvaa gbShRg header"
        >
          <img
            class="sc-gsDJrp dWTmlB"
            data-testid="logo"
            src="nasa.svg"
          />
          <span
            class="sc-dkPtyc fkhCXL"
          >
            MARS ROVER
          </span>
        </header>
        <main
          class="sc-jrQzUz flytEY"
        >
          <h1
            class="sc-iqsfdx loqvoP"
            style="margin-top: 3rem;"
          >
            ROVER CONTROL BOARD
          </h1>
          <div
            class="sc-crHlIS kwjHRW controlBoard"
          >
            <form
              class="sc-egiSv"
            >
              <div
                class="sc-dJjZJu hQJBly"
              >
                <div
                  class="sc-fFehDp sc-bkkfTU dSPLbK kdYexC"
                >
                  <label
                    class="sc-ksdxAp WIMUZ search__label"
                    for="inputControl"
                  >
                    Send order to rover:
                  </label>
                </div>
                <div
                  class="sc-fFehDp sc-ieebsP dSPLbK hvXGLB"
                  style="display: flex; justify-content: space-between;"
                >
                  <input
                    class="sc-bqiQRQ fzjAig"
                    id="inputControl"
                    type="text"
                    value=""
                  />
                  <button
                    class="sc-fotPbf cnbLHY"
                    disabled=""
                    type="submit"
                  >
                    Go!
                  </button>
                </div>
              </div>
              <div
                class="sc-dJjZJu hQJBly"
              >
                <div
                  class="sc-fFehDp sc-bkkfTU dSPLbK kdYexC"
                >
                  <label
                    class="sc-ksdxAp WIMUZ"
                  >
                    Rover direction:
                  </label>
                </div>
                <div
                  class="sc-fFehDp sc-ieebsP dSPLbK hvXGLB"
                >
                  <p
                    class="sc-hBURRC bKFbtw"
                  >
                     
                    N
                  </p>
                </div>
              </div>
            </form>
            <div
              class="sc-dJjZJu hQJBly"
            >
              <div
                class="sc-fFehDp sc-bkkfTU dSPLbK kdYexC"
              >
                <label
                  class="sc-ksdxAp WIMUZ"
                >
                   Rover position:
                </label>
              </div>
              <div
                class="sc-fFehDp sc-ieebsP dSPLbK hvXGLB"
                style="display: flex; justify-content: space-between;"
              >
                <label
                  class="sc-ksdxAp WIMUZ"
                >
                  X:0 Y:0
                </label>
                <button
                  class="sc-fotPbf cnbLHY"
                >
                  Reset
                </button>
              </div>
            </div>
          </div>
          
          <div
            class="sc-iCfLBT jtplsD grid"
          >
            <div
              class="sc-furvIG iZMSZS"
              data-testid="99"
              id="99"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="89"
              id="89"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="79"
              id="79"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="69"
              id="69"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="59"
              id="59"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="49"
              id="49"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="39"
              id="39"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="29"
              id="29"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="19"
              id="19"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="09"
              id="09"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="98"
              id="98"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="88"
              id="88"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="78"
              id="78"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="68"
              id="68"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="58"
              id="58"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="48"
              id="48"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="38"
              id="38"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="28"
              id="28"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="18"
              id="18"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="08"
              id="08"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="97"
              id="97"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="87"
              id="87"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="77"
              id="77"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="67"
              id="67"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="57"
              id="57"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="47"
              id="47"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="37"
              id="37"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="27"
              id="27"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="17"
              id="17"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="07"
              id="07"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="96"
              id="96"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="86"
              id="86"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="76"
              id="76"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="66"
              id="66"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="56"
              id="56"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="46"
              id="46"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="36"
              id="36"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="26"
              id="26"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="16"
              id="16"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="06"
              id="06"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="95"
              id="95"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="85"
              id="85"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="75"
              id="75"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="65"
              id="65"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="55"
              id="55"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="45"
              id="45"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="35"
              id="35"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="25"
              id="25"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="15"
              id="15"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="05"
              id="05"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="94"
              id="94"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="84"
              id="84"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="74"
              id="74"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="64"
              id="64"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="54"
              id="54"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="44"
              id="44"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="34"
              id="34"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="24"
              id="24"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="14"
              id="14"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="04"
              id="04"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="93"
              id="93"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="83"
              id="83"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="73"
              id="73"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="63"
              id="63"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="53"
              id="53"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="43"
              id="43"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="33"
              id="33"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="23"
              id="23"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="13"
              id="13"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="03"
              id="03"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="92"
              id="92"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="82"
              id="82"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="72"
              id="72"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="62"
              id="62"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="52"
              id="52"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="42"
              id="42"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="32"
              id="32"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="22"
              id="22"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="12"
              id="12"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="02"
              id="02"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="91"
              id="91"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="81"
              id="81"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="71"
              id="71"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="61"
              id="61"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="51"
              id="51"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="41"
              id="41"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="31"
              id="31"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="21"
              id="21"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="11"
              id="11"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="01"
              id="01"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="90"
              id="90"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="80"
              id="80"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="70"
              id="70"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="60"
              id="60"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="50"
              id="50"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="40"
              id="40"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="30"
              id="30"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="20"
              id="20"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="10"
              id="10"
            />
            <div
              class="sc-furvIG iZMSZS"
              data-testid="00"
              id="00"
            >
              <img
                alt="rover"
                class="sc-pVTma ghyLA-d"
                direction="N"
                src="rover.png"
              />
            </div>
          </div>
        </main>
        <div
          class="sc-hKwCoD CQaOl footer"
        >
          <footer
            class="sc-eCImvq PvPWG"
          >
            <a
              class="footer__link"
              data-testid="github-link"
              href="https://github.com/stephane777"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                class="sc-gKckTs lehCFg"
                data-testid="github-svg"
              >
                <use
                  class="grid__icon-use"
                  href="sprite.svg#icon-github"
                />
              </svg>
            </a>
            <p
              class="sc-jRQAMF fvfIEH"
            >
              Created by Stephane Candelas.
            </p>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="sc-bdvvaa gbShRg header"
      >
        <img
          class="sc-gsDJrp dWTmlB"
          data-testid="logo"
          src="nasa.svg"
        />
        <span
          class="sc-dkPtyc fkhCXL"
        >
          MARS ROVER
        </span>
      </header>
      <main
        class="sc-jrQzUz flytEY"
      >
        <h1
          class="sc-iqsfdx loqvoP"
          style="margin-top: 3rem;"
        >
          ROVER CONTROL BOARD
        </h1>
        <div
          class="sc-crHlIS kwjHRW controlBoard"
        >
          <form
            class="sc-egiSv"
          >
            <div
              class="sc-dJjZJu hQJBly"
            >
              <div
                class="sc-fFehDp sc-bkkfTU dSPLbK kdYexC"
              >
                <label
                  class="sc-ksdxAp WIMUZ search__label"
                  for="inputControl"
                >
                  Send order to rover:
                </label>
              </div>
              <div
                class="sc-fFehDp sc-ieebsP dSPLbK hvXGLB"
                style="display: flex; justify-content: space-between;"
              >
                <input
                  class="sc-bqiQRQ fzjAig"
                  id="inputControl"
                  type="text"
                  value=""
                />
                <button
                  class="sc-fotPbf cnbLHY"
                  disabled=""
                  type="submit"
                >
                  Go!
                </button>
              </div>
            </div>
            <div
              class="sc-dJjZJu hQJBly"
            >
              <div
                class="sc-fFehDp sc-bkkfTU dSPLbK kdYexC"
              >
                <label
                  class="sc-ksdxAp WIMUZ"
                >
                  Rover direction:
                </label>
              </div>
              <div
                class="sc-fFehDp sc-ieebsP dSPLbK hvXGLB"
              >
                <p
                  class="sc-hBURRC bKFbtw"
                >
                   
                  N
                </p>
              </div>
            </div>
          </form>
          <div
            class="sc-dJjZJu hQJBly"
          >
            <div
              class="sc-fFehDp sc-bkkfTU dSPLbK kdYexC"
            >
              <label
                class="sc-ksdxAp WIMUZ"
              >
                 Rover position:
              </label>
            </div>
            <div
              class="sc-fFehDp sc-ieebsP dSPLbK hvXGLB"
              style="display: flex; justify-content: space-between;"
            >
              <label
                class="sc-ksdxAp WIMUZ"
              >
                X:0 Y:0
              </label>
              <button
                class="sc-fotPbf cnbLHY"
              >
                Reset
              </button>
            </div>
          </div>
        </div>
        
        <div
          class="sc-iCfLBT jtplsD grid"
        >
          <div
            class="sc-furvIG iZMSZS"
            data-testid="99"
            id="99"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="89"
            id="89"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="79"
            id="79"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="69"
            id="69"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="59"
            id="59"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="49"
            id="49"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="39"
            id="39"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="29"
            id="29"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="19"
            id="19"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="09"
            id="09"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="98"
            id="98"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="88"
            id="88"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="78"
            id="78"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="68"
            id="68"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="58"
            id="58"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="48"
            id="48"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="38"
            id="38"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="28"
            id="28"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="18"
            id="18"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="08"
            id="08"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="97"
            id="97"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="87"
            id="87"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="77"
            id="77"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="67"
            id="67"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="57"
            id="57"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="47"
            id="47"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="37"
            id="37"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="27"
            id="27"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="17"
            id="17"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="07"
            id="07"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="96"
            id="96"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="86"
            id="86"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="76"
            id="76"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="66"
            id="66"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="56"
            id="56"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="46"
            id="46"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="36"
            id="36"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="26"
            id="26"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="16"
            id="16"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="06"
            id="06"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="95"
            id="95"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="85"
            id="85"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="75"
            id="75"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="65"
            id="65"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="55"
            id="55"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="45"
            id="45"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="35"
            id="35"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="25"
            id="25"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="15"
            id="15"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="05"
            id="05"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="94"
            id="94"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="84"
            id="84"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="74"
            id="74"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="64"
            id="64"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="54"
            id="54"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="44"
            id="44"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="34"
            id="34"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="24"
            id="24"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="14"
            id="14"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="04"
            id="04"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="93"
            id="93"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="83"
            id="83"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="73"
            id="73"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="63"
            id="63"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="53"
            id="53"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="43"
            id="43"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="33"
            id="33"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="23"
            id="23"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="13"
            id="13"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="03"
            id="03"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="92"
            id="92"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="82"
            id="82"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="72"
            id="72"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="62"
            id="62"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="52"
            id="52"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="42"
            id="42"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="32"
            id="32"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="22"
            id="22"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="12"
            id="12"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="02"
            id="02"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="91"
            id="91"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="81"
            id="81"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="71"
            id="71"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="61"
            id="61"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="51"
            id="51"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="41"
            id="41"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="31"
            id="31"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="21"
            id="21"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="11"
            id="11"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="01"
            id="01"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="90"
            id="90"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="80"
            id="80"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="70"
            id="70"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="60"
            id="60"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="50"
            id="50"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="40"
            id="40"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="30"
            id="30"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="20"
            id="20"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="10"
            id="10"
          />
          <div
            class="sc-furvIG iZMSZS"
            data-testid="00"
            id="00"
          >
            <img
              alt="rover"
              class="sc-pVTma ghyLA-d"
              direction="N"
              src="rover.png"
            />
          </div>
        </div>
      </main>
      <div
        class="sc-hKwCoD CQaOl footer"
      >
        <footer
          class="sc-eCImvq PvPWG"
        >
          <a
            class="footer__link"
            data-testid="github-link"
            href="https://github.com/stephane777"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              class="sc-gKckTs lehCFg"
              data-testid="github-svg"
            >
              <use
                class="grid__icon-use"
                href="sprite.svg#icon-github"
              />
            </svg>
          </a>
          <p
            class="sc-jRQAMF fvfIEH"
          >
            Created by Stephane Candelas.
          </p>
        </footer>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
